win32ss/user/ntuser/winpos.c

```diff
@@ -3391,36 +3392,30 @@ NtUserGetWindowPlacement(HWND hWnd,
 {
    PWND Wnd;
    WINDOWPLACEMENT Safepl;
-   NTSTATUS Status;
    BOOL Ret = FALSE;
 
    TRACE("Enter NtUserGetWindowPlacement\n");
    UserEnterShared();
 
    if (!(Wnd = UserGetWindowObject(hWnd)))
-   {
       goto Exit; // Return FALSE
-   }
 
-   Status = MmCopyFromCaller(&Safepl, lpwndpl, sizeof(WINDOWPLACEMENT));
-   if (!NT_SUCCESS(Status))
+   _SEH2_TRY
    {
-      SetLastNtError(Status);
-      goto Exit; // Return FALSE
+      ProbeForRead(&lpwndpl->length, sizeof(lpwndpl->length), 1);
+      Safepl.length = lpwndpl->length; // Use length value
```

Comment:
Wouldn't it be of interest to also validate the value of length here as well?

Reply:
`IntGetWindowPlacement` will do it.